packages/transport-tcp/test/socket-to-conn.spec.ts

```diff
@@ -287,6 +289,52 @@ describe('socket-to-conn', () => {
     expect(serverSocket.destroyed).to.be.true()
   })
 
+  it('should not close MultiaddrConnection twice', async () => {
+    const loggerStub = Sinon.stub()
+    const logger: ComponentLogger = {
+      forComponent: () => loggerStub as unknown as Logger
+    };
+    ({ server, clientSocket, serverSocket } = await setup())
+
+    // promise that is resolved when our outgoing socket is closed
```

Comment:
Add a spy on `.destroy()` so we can assert that the socket was only closed once:

```suggestion

    // spy on `.destroy()` invocations
    const serverSocketDestroySpy = Sinon.spy(serverSocket, 'destroy')

    // promise that is resolved when our outgoing socket is closed
```